core/vm/logger.go

```diff
@@ -200,6 +206,72 @@ func (l *StructLogger) CaptureState(pc uint64, op OpCode, gas, cost uint64, scop
 			)
 			l.storage[contract.Address()][address] = value
 			storage = l.storage[contract.Address()].Copy()
+
+			proof, err = l.env.StateDB.GetStorageProof(contract.Address(), address)
+			if err != nil {
+				log.Warn("Failed to get proof", "contract address", contract.Address().String(), "key", address.String(), "err", err)
+			}
+		}
+	}
+	var rdata []byte
+	if l.cfg.EnableReturnData {
+		rdata = make([]byte, len(rData))
+		copy(rdata, rData)
+	}
+	// create a new snapshot of the EVM.
+	log := StructLog{pc, op, gas, cost, mem, memory.Len(), stck, rdata, proof, storage, depth, l.env.StateDB.GetRefund(), err}
+	l.logs = append(l.logs, log)
+}
+
+// CaptureStateSpecial for special needs, tracks SSTORE ops and records the storage change.
+func (l *StructLogger) CaptureStateSpecial(pc uint64, op OpCode, gas, cost uint64, scope *ScopeContext, rData []byte, depth int, err error) {
+	if op != SSTORE {
+		return
+	}
+	memory := scope.Memory
+	stack := scope.Stack
+	contract := scope.Contract
+	// check if already accumulated the specified number of logs
+	if l.cfg.Limit != 0 && l.cfg.Limit <= len(l.logs) {
+		return
+	}
+	// Copy a snapshot of the current memory state to a new buffer
+	var mem []byte
+	if l.cfg.EnableMemory {
+		mem = make([]byte, len(memory.Data()))
+		copy(mem, memory.Data())
+	}
+	// Copy a snapshot of the current stack state to a new buffer
+	var stck []uint256.Int
+	if !l.cfg.DisableStack {
+		stck = make([]uint256.Int, len(stack.Data()))
+		for i, item := range stack.Data() {
+			stck[i] = item
+		}
+	}
+	// Copy a snapshot of the current storage to a new container
+	var (
+		storage Storage
+		pprof   [][]byte
+	)
+	if !l.cfg.DisableStorage && stack.len() >= 2 {
+		// initialise new changed values storage container for this contract
+		// if not present.
+		if l.storage[contract.Address()] == nil {
+			l.storage[contract.Address()] = make(Storage)
+		}
+		// capture SLOAD opcodes and record the read entry in the local storage
```

Comment:
according to your
```
	if op != SSTORE {
		return
	}
```

we cannot capture `SLOAD` here

Reply:
SLOAD won't change storage, do we need add that?